pkg/cmd/admin/router/router.go

```diff
@@ -225,6 +230,15 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			return fmt.Errorf("router %q does not exist (no service)", name)
 		}
 
+		if len(cfg.ServiceAccount) == 0 {
+			return fmt.Errorf("router could not be created; you must specify a service account with access to host ports with --service-account")
```

Comment:
I don't think this error message should say anything about host ports -- the problem is that no service account is specified